The contribution guidelines (#38)
AckeeCZ/frontend-cookbook: pages/contribution/Guidelines.md

# How can I contribute?

The guidelines for balance between innovation and code stability.

---

> > Ok, I have a new idea. Now, what's the right way to implement it to move FE in Ackee forward and let other team members have a benefit from that?

Reviewer: ```suggestion
> Ok, I have a new idea. Now, what's the right way to implement it to move FE in Ackee forward and let other team members benefit from that?
```

1. 💡**Idea validation**

   New ideas are very much welcomed. However, the team has to maintain its solid base that is built on. Therefore, the new idea must respect:

   - [team values and goals](https://trello.com/b/tnr6JKtk/c%C3%ADle-vize-a-hodnoty-t%C3%BDmu),
   - [our stack [To Do]](/pages/application-stack),
   - [our good practices [To Do]](/pages/BestPractices.html) and rest of this frontend cookbook.

2. ⏳**Rought time estimate**

>    Next step is make at least rought time estimate:

Reviewer: ```suggestion
   Next step is to make at least a rough time estimate:
```

   - <5h - No need to get any approval.
   - 5-20h - An approval by your team leader required.
   - \>20h - An approval from with higher management required.

3. 🤓**Consultation with expert**

>    Consider to discuss your idea with team expert for that given thing based on the [Responsible matrix](https://frontend.ack.ee/responsibility-matrix). You might save a lot of time to yourself by receiving valuable feedback early on.

Reviewer: ```suggestion
   Consider discussing your idea with a team expert for that given thing based on the [Responsible matrix](https://frontend.ack.ee/responsibility-matrix). You might save a lot of time to yourself by receiving valuable feedback early on.
```

4. ✅**The things that should be included**

>    - A **documentation** that can each member to understand. Otherwise barely anyone will use it then. The form can be readme, next cookbook chapter or fully featured documentation with its own domain.

Reviewer: ```suggestion
   - A **documentation**  each member can understand. Otherwise barely anyone will use it then. The form can be readme, next cookbook chapter or fully featured documentation with its own domain.
```

   - **Tests** are heavily recommended. Don't forget it'll be your new responsibility, so this will save you time in the future.

5. 🧑‍🏫**Presenting it to the team**

   Remember that other team members have to use it and first of all have to know how to use it.

>    Even though sharing the docs with your team could seem to be a good idea, imagine how many Slack messages you've ignored or missed to read. So maybe, it'd better idea to introduce it as [another topic during weekly FE meetings](https://frontend.ack.ee/).

Reviewer: ```suggestion
   Even though sharing the docs with your team could seem to be a good idea, imagine how many Slack messages you've ignored or missed. So maybe, it'd be a better idea to introduce it as [another topic during weekly FE meetings](https://frontend.ack.ee/).
```

Very good point by the way 👍

6. 🧑‍💻**Use it on a client project**

   Now, the only thing missing is to find the right time to use it on client projects.

>    The beginning or some major breaking point could be the right time. Definitly, it wouldn't be a good idea to switch on that during a running sprint.

Reviewer 2: Definitely
